zarr/util.py

```diff
@@ -696,3 +701,28 @@ def all_equal(value: Any, array: Any):
             # using == raises warnings from numpy deprecated pattern, but
             # using np.equal() raises type errors for structured dtypes...
             return np.all(value == array)
+
+
+def ensure_contiguous_ndarray_or_bytes(buf) -> Union[NDArrayLike, bytes]:
```

Comment:
We may want to move this to [`numcodecs.compat`]( https://github.com/zarr-developers/numcodecs/blob/67ede4c6b4f1707f1da18351945e10904a5572de/numcodecs/compat.py ), but that can be follow-on work